### ISSUE TITLE
Allow jsBeautify to be used for XML

### ISSUE THREAD

**Issue by moreginger** [Glavin001/atom-beautify#369]
Beautification of XML used to use jsBeautify and it worked OK (apart from https://github.com/beautify-web/js-beautify/pull/683), in particular it has better options (then PrettyDiff) for attribute wrapping etc. Could it be made a settings option again.

**Comment by prettydiff** [comment]
Could you provide a code sample that you believe to be in error so that I may work on improving this for you?

Pretty Diff was chosen over jsBeautify because jsBeautify uses an HTML parser to provide incomplete support for XML. Officially, jsBeautify project does not support XML at all, though it sounds like they are work on this. The incomplete and hacky support for XML made it difficult to extend base XML support for various edge cases, templating schemes, and particuarly React JSX format. The straw that broke the camel's back for this project was an internationalization issue.

It is important to note that jsBeautify now provides for React JSX using a ES4X library to provide some JavaScript to XML conformance. Both projects based their support off this document that I assembled: https://github.com/prettydiff/prettydiff/blob/master/unit_tests/beautification_javascript_jsx.txt

The Pretty Diff approach is to provide a base markup parser than apply different rules for various different language formats. The Pretty Diff approach is not necessarily better or more stable but is certainly more extensible.

**Comment by moreginger** [comment]
Thanks for the detailed answer!
Notably in jsBeautify you can force wrap one attribute per line to get something like:

```xml
<foo xmlns:ns1="ns1"
  xmlns:ns2="ns2"
  xmlns:ns3="ns3" />
```

...which can be easier to work with. It also allows the attribute indent size to be set separately to the standard indent.

I'm not sure whether I prefer the first attribute appearing on the same line as the tag name. I got used to it recently when using atom-beautifier/jsBeautifier, but the pretty printer I have previously used also put the first attribute on a separate line.

**Comment by Glavin001** [comment]
Since this issue is add XML language to JS-Beautify beautifier, here are instructions for what would need to be done and submitted in a Pull Request. It's very straight forward:
- [ ] Add `XML: true` to JS-Beautify `options` at https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/js-beautify.coffee#L7 This will allow JS Beautify be be selectable as a beautifier for language XML, by going to the package options.
- [ ] Set default beautifier for XML. Since `Pretty Diff` has better support in most cases, I'm going to recommend it and want it to be the default. Create field `defaultBeautifier` in https://github.com/Glavin001/atom-beautify/blob/master/src/languages/xml.coffee#L23 and set to `Pretty Diff`. See https://github.com/Glavin001/atom-beautify/blob/master/src/languages/coffeescript.coffee#L21 for example.

I will accept a pull request with the above implemented. Thanks!

**Comment by Glavin001** [comment]
Published to v0.28.1
